mod_auth/controllers.py

```diff
@@ -108,6 +116,42 @@ def send_reset_email(usr):
               'error-message')
 
 
+@mod_auth.route('/github_redirect', methods=['GET', 'POST'])
+def github_redirect():
+    from run import config
+    github_clientid = config.get('GITHUB_CLIENT_ID', '')
+    return redirect(('https://github.com/login/oauth/authorize?'
+                     'client_id={client_id}&scope=public_repo').format(
+        client_id=github_clientid))
+
+
+@mod_auth.route('/github_callback', methods=['GET', 'POST'])
+@template_renderer()
+def github_callback():
+    from run import config
+    if 'code' in request.args:
+        url = 'https://github.com/login/oauth/access_token'
+        payload = {
+            'client_id': config.get('GITHUB_CLIENT_ID', ''),
```

Comment:
If new variables are added to the config, please add them to the config_sample.py!